### PR TITLE
Enable CoreNEURON CI with NMODL

### PR DESCRIPTION
**Pull request by pramodk**
Supersede #576 

Apply patch mentioned in https://github.com/neuronsimulator/nrn/pull/1261#issuecomment-866811604

CI_BRANCHES:NEURON_BRANCH=master,

### CODE REVIEW

**Comment by pramodk** [comment]
NEURON PR https://github.com/neuronsimulator/nrn/pull/1347 is pending but that is not a blocker for this PR.

@alkino : anything else needs to be done here?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -74,6 +85,21 @@ build:coreneuron:intel:
     - .spack_build
     - .spack_intel
 
+build:coreneuron+nmodl:gpu:
+  variables:
+    SPACK_PACKAGE: coreneuron
+    # +report pulls in a lot of dependencies and the tests fail.
+    # See https://github.com/BlueBrain/CoreNeuron/issues/518 re: build_type
+    SPACK_PACKAGE_SPEC: +nmodl+gpu+tests~legacy-unit~report build_type=RelWithDebInfo
+    SPACK_PACKAGE_DEPENDENCIES: ^bison%gcc^flex%gcc^eigen%gcc
```
Comment (alkino) [inline .gitlab-ci.yml:94]: Should delete it. It's handled by the `before_script` line 98.

Comment (olupton) [inline .gitlab-ci.yml:94]: I think this line can be dropped (its value will be overridden by the `dotenv` artifacts from the `build:nmodl:gpu` job).
```suggestion
```

**Review by olupton — APPROVED** [review]
LGTM, one line that should be removed to avoid confusion.

We could probably reduce repetition in the YAML file with a couple of extra templates (`.spack_coreneuron` and `.spack_nmodl` à la `.spack_neuron`, but it's not a big deal).

**Review by alkino — APPROVED** [review]

**Pull request merged**